version: '3.8'

services:
  db:
    image: postgres:12
    container_name: db
    environment:
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    ports:
      - "5432:5432"
    volumes:
      - ./axelor-data:/var/lib/postgresql/data
      - ./backup:/var/lib/postgresql/backup:rw
    restart: always
    networks:
      - axelor-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER}"]
      interval: 10s
      timeout: 5s
      retries: 5

  axelor-app:
    image: kapsiki/kapsiki-axelor:latest
    restart: always
    container_name: axelor-app
    depends_on:
      - db
    environment:
      AXELOR_CONFIG_DB_DEFAULT_URL: ${AXELOR_CONFIG_DB_DEFAULT_URL}
      AXELOR_CONFIG_DB_DEFAULT_USER: ${AXELOR_CONFIG_DB_DEFAULT_USER}
      AXELOR_CONFIG_DB_DEFAULT_PASSWORD: ${AXELOR_CONFIG_DB_DEFAULT_PASSWORD}
    ports:
      - "8080:8080"
    networks:
      - axelor-network
    healthcheck:
      test: ["CMD-SHELL", "curl -f http://localhost:8080/health"]
      interval: 10s
      timeout: 5s
      retries: 5

networks:
  axelor-network:
    driver: bridge
    ipam:
      config:
        - subnet: 172.10.2.0/24
